pkg/ingester/wal.go

```diff
@@ -186,6 +186,12 @@ func (w *walWrapper) Log(record *WALRecord) error {
 	case <-w.quit:
 		return nil
 	default:
+		if len(record.Series) == 0 && len(record.Samples) == 0 {
```

Comment:
Why the check is here? To me looks easier to understand if we move it up, right after `if record == nil`.